### PR TITLE
(PA-2028) Update to support running debian under systemd or sysv

### PR DESCRIPTION
**Pull request by underscorgan**
Needs https://github.com/puppetlabs/vanagon/pull/684 and vanagon release
Systemd check was updated to the one used in systemd service provider.

### CODE REVIEW

**Comment by underscorgan** [comment]
This is now supported in vanagon.

**Comment by branan** [comment]
Please move this to a PA ticket, rather than a VANAGON ticket, so we can track it in our releases

**Comment by branan** [comment]
I'm also still pretty disinclined to merge this until we figure out what our testing/supportability story is, FWIW

**Comment by McdonaldSeanp** [comment]
I agree with @branan, we need to determine what expectations are around our support in this situation from both us and customers.

We need to be able to identify if this scenario requires it's own row in test matrices and what that would look like before I'm comfortable merging

File: configs/components/pxp-agent.rb
Comment (mwaggett) [inline configs/components/pxp-agent.rb]: Why do we need this check if we're already in the "systemd" case?

**Comment by mwaggett** [comment]
Seems fine to me from a code perspective, but I will defer to @branan and @McdonaldSeanp for whether adding this is a good idea rn.

File: configs/components/pxp-agent.rb
Comment (underscorgan) [reply configs/components/pxp-agent.rb]: For debian platforms with both sysv and systemd you need to add all of the scripts, and then run the scripts based on what's currently on the system. You only know you're in the systemd section at build time, but this can change at runtime so it needs to be in the scripts.

Comment (mwaggett) [reply configs/components/pxp-agent.rb]: oookay! thanks!

Comment (olifre) [reply configs/components/pxp-agent.rb]: This broke automated deployment for us with Ubuntu, since `/run/systemd/system` does not exist in the installer environment, so services will not be activated anymore at all since this change.

Comment (underscorgan) [reply configs/components/pxp-agent.rb]: @olifre Is there a way in your environment to tell which init system it's running? We want to be able to support deb-ish systems running either sysv/upstart or systemd

Comment (olifre) [reply configs/components/pxp-agent.rb]: @underscorgan As outlined in the long-standing https://tickets.puppetlabs.com/browse/PA-136 , which is about the very same issue concerning runtime behaviour of the agent for the same very same failing test against the runtime directory, the correct / portable check appears to be to test whether `/sbin/init` links to `/lib/systemd/systemd`. This works in Ubuntu and Debian installer chroots, and I still have to see a systemd system where it fails ;-).

Comment (underscorgan) [reply configs/components/pxp-agent.rb]: Honestly, I don't remember why this was needed here. If this work is going to move forward I can investigate further, but I was working on this too long ago to remember edge cases I ran into while testing.

**Comment by ciprianbadescu** [comment]
passing ad-hoc: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/926/

**Review by gimmyxd — APPROVED** [review]

**Review by GabrielNagy — APPROVED** [review]
good to merge for me, after pxp-agent logrotate for launchd is readded

**Pull request merged**
